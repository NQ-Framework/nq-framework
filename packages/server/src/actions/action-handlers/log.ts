import {
  ActionHandler,
  ActionInstance,
  PropertyValue,
  WorkflowExecutionContext,
} from '@nqframework/models';
import { exception } from 'console';

export const handler: ActionHandler = {
  handle: async (
    propertyValues: PropertyValue[],
    actionInstance: ActionInstance,
    workflowExecution: WorkflowExecutionContext,
  ): Promise<PropertyValue[]> => {
    const message = propertyValues.find((i) => i.name === 'message')?.value;
    if (!message) {
      throw new Error('Missing required parameter message');
    }
    console.log(
      'Log action instance says :' +
      message +
      ' instance: ' +
      JSON.stringify(actionInstance) +
      '  context: ' +
      JSON.stringify(workflowExecution),
    );

    return [{
      name: 'message',
      value: message
    }];
  },
};
