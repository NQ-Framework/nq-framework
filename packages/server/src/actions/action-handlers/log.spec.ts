import { handler } from './log';

describe('log action', () => {
  it('should log message value and pass it back out', async () => {
    const result = await handler.handle(
      [{ name: 'message', value: 'test' }],
      {} as any,
      {} as any,
      {} as any,
    );
    expect(result).toEqual([{ name: 'message', value: 'test' }]);
  });
  it('should throw when message is not present', async () => {
    await expect(
      handler.handle(
        [{ name: 'not message', value: 'test' }],
        {} as any,
        {} as any,
        {} as any,
      ),
    ).rejects.toThrowErrorMatchingSnapshot();
  });
});
