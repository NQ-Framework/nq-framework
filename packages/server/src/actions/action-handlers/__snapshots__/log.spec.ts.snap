// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`log action should throw when message is not present 1`] = `"Missing required parameter message"`;
